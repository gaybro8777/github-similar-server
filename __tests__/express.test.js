/**
 * @file main
 * @author imcuttle
 * @date 2018/4/4
 */
const express = require('express')
const nps = require('path')
const request = require('supertest')

const { makeFixture } = require('./helper')
const middleware = require('../src/index')

describe('main', function() {
  const app = express()

  app.use('/case-1', middleware({ root: makeFixture('case-1') }))
  app.use('/case-404', middleware({ root: makeFixture('case-404') }))
  app.use('/case-readme', middleware({ root: makeFixture('case-readme') }))

  it('should static works when path not found', function(done) {
    request(app)
      .get('/case-1/not_found')
      // .expect('Content-Type', /json/)
      // .expect('Content-Length', '15')
      .expect(404)
      .end(function(err, res) {
        if (err) throw err
        done()
      })
  })

  it('should static works when path not found, but has 404.html', function(done) {
    request(app)
      .get('/case-404/not_found')
      .expect(function(res) {
        expect(res.text).toMatchSnapshot()
      })
      .expect(404)
      .end(done)
  })

  it('should return 302 when directory matched', function(done) {
    request(app)
      .get('/case-readme/readme')
      .expect(function(res) {
        expect(res.text).toMatchSnapshot()
      })
      .expect(302)
      .end(done)
  })

  it('should skip readme when index.html existed', function(done) {
    request(app)
      .get('/case-readme')
      .expect(function(res) {
        expect(res.text).toBe('index\n')
      })
      .expect(200)
      .end(done)
  })
})
