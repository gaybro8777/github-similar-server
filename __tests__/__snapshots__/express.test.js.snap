// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should return readme when existed 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <title>Readme</title>
  <link rel=\\"stylesheet\\" href=\\"//unpkg.com/github-markdown-css\\">
  <link rel=\\"stylesheet\\" href=\\"//unpkg.com/highlight.js/styles/github.css\\">
  <style>
    .markdown-body {
      max-width: 65%;
      margin: 40px auto 30px;
    }
    @media screen and (max-width: 1200px) {
      .markdown-body {
        max-width: 78%;
      }
    }

    @media screen and (max-width: 700px) {
      .markdown-body {
        max-width: 90%;
        margin: 20px auto 10px;
      }
    }
    @media screen and (max-width: 400px) {
      .markdown-body {
        max-width: 98%;
        margin: 20px auto 10px;
      }
    }
  </style>
</head>
<body>
<div class=\\"markdown-body\\">
  <p>??
// haha</p>
</div>
</body>
</html>
"
`;

exports[`main should static works when path not found, but has 404.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
    <meta charset=\\"UTF-8\\">
    <title>404</title>
</head>
<body>

</body>
</html>
"
`;
